### DISCUSSION TITLE
e.remove is not a function

### DISCUSSION THREAD

**Discussion by hasanm95** [gatsbyjs/gatsby#36247] [Help]
### Why am I getting this issue? I am getting this issue while I try to route

`
One unhandled runtime error found in your files. See the list below to fix it:

Error in function eval in ./.cache/head/head-export-handler-for-browser.js:18
e.remove is not a function

./.cache/head/head-export-handler-for-browser.js:18
Open in Editor
  16 | const removePrevHeadElements = () => {
  17 |   const prevHeadNodes = [...document.querySelectorAll(`[data-gatsby-head]`)]
> 18 |   prevHeadNodes.forEach(e => e.remove())
     |                               ^
  19 | }
  20 |
  21 | const onHeadRendered = () => {`

**Comment by pieh** [comment]
This looks like a bug - can you open a bug report ( https://github.com/gatsbyjs/gatsby/issues/new?assignees=&labels=type%3A+bug&template=BUG_REPORT.yml ) ideally with reproduction?

**Reply by hasanm95** [reply]
"@babel/preset-env" was causing this issue. After removing it everything is working fine now

**Comment by BArtistically** [comment]
The problem with me was  also in .babelrc for "@babel/preset-env", the setting
 "loose": true 
that was causing this. 
It will make [...document.querySelectorAll(`[data-gatsby-head]`)] an Array with one element, the nodelist object.
Without the loose it will convert the nodelist to an array as it should.

**Comment by sidyes** [comment]
Thanks for the babel hint, I am facing the same issue in NRWL NX monorepo having gatsby projects.
A temporary solution is to use the following `.babelrc` config:

```json
{
  "presets": [
    [
      "@nrwl/gatsby/babel",
      {
        "reactRuntime": "automatic"
      }
    ],
    [
      "@babel/preset-env",
      {
        "loose": false // needed for Gatsby HEAD API
      }
    ]
  ],
  "plugins": [
    ["@babel/plugin-proposal-class-properties", { "loose": true }], // disables warning
    ["@babel/plugin-proposal-private-methods", { "loose": true }], // disables warning
    ["@babel/plugin-proposal-private-property-in-object", { "loose": true }] // disables warning
  ]
}
```

But the better approach would be to improve the implementation on Gatsby's Heads API side, right? Can you please improve it try/catch or ignore such  cases like described by @BArtistically ?
